setup.py

```diff
@@ -2,433 +2,517 @@
+            import torch
+        except ImportError:
+            pass
+        else:
+            _frameworks.append("pytorch")
+        try:
+            import jax
+        except ImportError:
+            pass
+        else:
+            _frameworks.append("jax")
+        try:
+            import tensorflow
+        except ImportError:
+            pass
+        else:
+            _frameworks.append("tensorflow")
+
+    # Special framework names
+    if "all" in _frameworks:
+        _frameworks = supported_frameworks.copy()
+    if "none" in _frameworks:
+        _frameworks = []
+
+    # Check that frameworks are valid
+    _frameworks = [framework.lower() for framework in _frameworks]
+    for framework in _frameworks:
+        if framework not in supported_frameworks:
+            raise ValueError(
+                f"Transformer Engine does not support framework={framework}"
+            )
 
-    @staticmethod
-    def install_requires():
-        return []
-
-class PyTorchBuilder(FrameworkBuilderBase):
-    def __init__(self, *args, **kwargs) -> None:
-        pytorch_args = copy.deepcopy(args)
-        pytorch_kwargs = copy.deepcopy(kwargs)
-        from torch.utils.cpp_extension import BuildExtension
-        self.pytorch_build_extensions = BuildExtension(*pytorch_args, **pytorch_kwargs)
-
-    def initialize_options(self):
-        self.pytorch_build_extensions.initialize_options()
-
-    def finalize_options(self):
-        self.pytorch_build_extensions.finalize_options()
-
-    def run(self, extensions):
-        other_ext = [
-            ext for ext in extensions if not isinstance(ext, CMakeExtension)
+    return _frameworks
+
+# Call once in global scope since this function manipulates the
+# command-line arguments. Future calls will use a cached value.
+frameworks()
+
+def setup_requirements() -> Tuple[List[str], List[str], List[str]]:
+    """Setup Python dependencies
+
+    Returns dependencies for build, runtime, and testing.
+
+    """
+
+    # Common requirements
+    setup_reqs: List[str] = []
+    install_reqs: List[str] = ["pydantic"]
+    test_reqs: List[str] = ["pytest"]
+
+    def add_unique(l: List[str], vals: Union[str, List[str]]) -> None:
+        """Add entry to list if not already included"""
+        if isinstance(vals, str):
+            vals = [vals]
+        for val in vals:
+            if val not in l:
+                l.append(val)
+
+    # Requirements that may be installed outside of Python
+    if not found_cmake():
+        add_unique(setup_reqs, "cmake>=3.18")
+    if not found_ninja():
+        add_unique(setup_reqs, "ninja")
+
+    # Framework-specific requirements
+    if "pytorch" in frameworks():
+        add_unique(install_reqs, ["torch", "flash-attn>=1.0.2"])
+        add_unique(test_reqs, ["numpy", "onnxruntime", "torchvision"])
+    if "jax" in frameworks():
+        if not found_pybind11():
+            add_unique(setup_reqs, "pybind11")
+        add_unique(install_reqs, ["jax", "flax"])
```

Comment:
I forgot, but isn't flax an optional dependency now?
If so, we need to move it to tests.

Reply:
The plan is to make Flax optional, but right now we still import the Flax modules into the `transformer_engine.jax` namespace for backwards-compatibility. I think it would be reasonable if we modified that so we skipped those imports if Flax is not detected.

Reply:
Lets do that in another PR.